controllers/rollingupgrade_controller.go

```diff
@@ -112,7 +110,7 @@ type RollingUpgradeReconciler struct {
 	EC2Client       ec2iface.EC2API
 	ASGClient       autoscalingiface.AutoScalingAPI
 	generatedClient *kubernetes.Clientset
-	Asg             *autoscaling.Group
+	Asgs            map[string]*autoscaling.Group
```

Comment:
I wonder if we can get rid of this altogether and switch everything to using `ruObjNameToASG` which is already a sync.Map.
We just need to make sure we overwriting objects when we call `populateAsg`